Makefile

```
@@ -28,239 +50,258 @@ else
+# These images MUST exist in a registry and be pull-able.
+BUNDLE_IMGS ?= $(BUNDLE_IMG)
 
-prepare-alpha-release: bump-release generate fmt vet manifests bundle
+# The image tag given to the resulting catalog image (e.g. make catalog-build CATALOG_IMG=example.com/operator-catalog:v0.2.0).
+CATALOG_IMG ?= $(IMAGE_TAG_BASE)-catalog:v$(VERSION)
 
-prepare-stable-release: bump-release generate fmt vet manifests bundle
-	$(MAKE) bundle CHANNELS=alpha,stable DEFAULT_CHANNEL=alpha
+# Default catalog base image to append bundles to
+CATALOG_BASE_IMG ?= $(IMAGE_TAG_BASE)-catalog:latest
+
+# Set CATALOG_BASE_IMG to an existing catalog image tag to add $BUNDLE_IMGS to that image.
+ifneq ($(origin CATALOG_BASE_IMG), undefined)
+FROM_INDEX_OPT := --from-index $(CATALOG_BASE_IMG)
+endif
+
+# Build a catalog image by adding bundle images to an empty catalog using the operator package manager tool, 'opm'.
+# This recipe invokes 'opm' in 'semver' bundle add mode. For more information on add modes, see:
+# https://github.com/operator-framework/community-operators/blob/7f1438c/docs/packaging-operator.md#updating-your-existing-operator
+.PHONY: catalog-build
+catalog-build: opm ## Build a catalog image.
+	$(OPM) index add --container-tool docker --mode semver --tag $(CATALOG_IMG) --bundles $(BUNDLE_IMGS) $(FROM_INDEX_OPT)
 
-bump-release:
-	sed -i.bak 's/version string = "v\(.*\)"/version string = "v$(VERSION)"/g' pkg/version/version.go \
-		&& rm pkg/version/version.go.bak
+# Push the catalog image.
+.PHONY: catalog-push
+catalog-push: ## Push a catalog image.
+	$(MAKE) docker-push IMG=$(CATALOG_IMG)
 
-bundle-push:
-	docker push $(BUNDLE_IMG)
+##@ Release
 
-catalog-build: $(OPM)
-	$(OPM) index add \
-		--build-tool docker \
-		--mode semver \
-		--bundles $(BUNDLE_IMG) \
-		--from-index $(CATALOG_IMG) \
-		--tag $(CATALOG_IMG)
+prepare-alpha-release: bump-release generate fmt vet manifests assets bundle ## Generates bundle manifests for alpha channel release
 
-catalog-push:
-	docker push $(CATALOG_IMG)
+prepare-stable-release: bump-release generate fmt vet manifests assets bundle refdocs ## Generates bundle manifests for stable channel release
+	$(MAKE) bundle CHANNELS=alpha,stable DEFAULT_CHANNEL=alpha
 
-bundle-publish: bundle-build bundle-push catalog-build catalog-push
+bump-release: ## Write release name to "pkg/version" package
+	sed -i 's/version string = "v\(.*\)"/version string = "v$(VERSION)"/g' pkg/version/version.go
 
+bundle-publish: bundle-build bundle-push catalog-build catalog-push catalog-retag-latest ## Generates and pushes all required images for a release
 
 get-new-release:
 	@hack/new-release.sh v$(VERSION)
 
-############################################
-#### Targets to manually test with Kind ####
-############################################
+catalog-retag-latest:
+	docker tag $(CATALOG_IMG) $(IMAGE_TAG_BASE)-catalog:latest
+	$(MAKE) docker-push IMG=$(IMAGE_TAG_BASE)-catalog:latest
```

You already have the var, instead of recreating it, let's use the one you have defiend above, isnt it?
```suggestion
	docker tag $(CATALOG_IMG) $(CATALOG_BASE_IMG)
	$(MAKE) docker-push IMG=$(CATALOG_BASE_IMG)
```